### PR TITLE
python312Packages.bidict: minor fixes

### PR DESCRIPTION
**Pull request by jab**
I am the author and maintainer of bidict. I noticed some issues with the nix package for it and thought I'd contribute some fixes (see below). This is my first PR to nixpkgs and I hope to contribute more in the future.

## Description of changes

* Correct minimum supported Python version.

  Bidict 0.23.1 (which nixpkgs recently updated to) dropped support for Python < 3.8.

  Ref: https://bidict.readthedocs.io/changelog.html

* Skip unnecessary integration tests.

  * Don't build and test the Sphinx docs or benchmarks, as this is overkill.
  * Drop associated dependencies from `nativeCheckInputs` now that they're not needed.

* Drop `py` dependency from nativeCheckInputs. `pytest` no longer depends on `py`.

  Ref: https://docs.pytest.org/en/7.4.x/changelog.html#deprecations

* Fix metadata:

  * Update outdated description string.

  * Use better homepage and changelog links.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/bidict/default.nix
Comment (jab) [inline pkgs/development/python-modules/bidict/default.nix:46]: (As of https://github.com/jab/bidict/commit/f48bf580dc0a49bf74e6085b4d349feeb180d46d I've removed the `--benchmark-*` options from pytest.ini, which will help avoid this once those changes are released.)

File: pkgs/development/python-modules/bidict/default.nix
Comment (jab) [inline pkgs/development/python-modules/bidict/default.nix:60]: @jbaum98, I'd be happy to add myself as a maintainer here in a followup PR if you'd like a hand.

File: pkgs/development/python-modules/bidict/default.nix
Comment (mweinelt) [inline pkgs/development/python-modules/bidict/default.nix:35]: We already provide `pytest` through the `pytestCheckHook`.

Comment (jab) [reply pkgs/development/python-modules/bidict/default.nix]: Thanks for the quick response and for catching that! Removed `pytest` in the latest revision. Is this now good to merge?

**Review by mweinelt — APPROVED** [review]
LGTM. Waiting for CI to finish.

**Comment by jab** [comment]
Thanks @mweinelt! Now seeing "All checks have passed".

**Pull request merged**
